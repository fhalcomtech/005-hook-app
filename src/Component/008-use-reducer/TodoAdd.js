import React from 'react';
import {useForm} from "../../hooks/useForm";

const TodoAdd = ({handlerSaveTodo}) => {
    const [formData, handlerInputChange, resetForm] = useForm({todoDesc: ""});
    const {todoDesc} = formData;
    const handlerSubmit = (e) => {
        e.preventDefault();
        if(todoDesc && todoDesc.length > 3 ) {
            const newTodo = {
                id: new Date().getTime(),
                desc: todoDesc,
                done: false,
            }
            handlerSaveTodo(newTodo);
            resetForm();
        }
    };

    return (
        <div className="col-4">

            <h2>Add Todo</h2>
            <hr/>
            <form onSubmit={handlerSubmit}>
                <input
                    type="text"
                    className="form-control"
                    placeholder="write todo description"
                    name="todoDesc"
                    value={todoDesc}
                    onChange={handlerInputChange}
                />
                <button className='btn btn-success form-control mt-2 ms-0' type="submit">Add</button>
            </form>
        </div>
    );
};

export default TodoAdd;
