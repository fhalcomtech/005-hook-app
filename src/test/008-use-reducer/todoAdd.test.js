import React from "react";
import { shallow } from "enzyme"
import TodoAdd from "../../Component/008-use-reducer/TodoAdd";

describe('Test TodoAdd', () => 
{ 
    const handlerSaveTodo = jest.fn();
    const wrapper = shallow(<TodoAdd handlerSaveTodo={handlerSaveTodo}/>);
    test('Test TodoAdd create snapshot', () => { 
        expect(wrapper).toMatchSnapshot();

     });

     test('Test TodoAdd Should not to be call handlerSavetodo', () => { 
         const form = wrapper.find("form");
         form.simulate("submit",{preventDefault:()=>{}});
         expect(handlerSaveTodo).toBeCalledTimes(0);
      })

      test('Test TodoAdd Should not to be call handlerSavetodo', (done) => { 
        const newTodoDesc = "Learn Vue Js";
        wrapper.find('[name="todoDesc"]').simulate("change", {target: {value:newTodoDesc, name: 'todoDesc'}})
        wrapper.find("form").simulate("submit",{preventDefault:()=>{}});
        expect(handlerSaveTodo).toBeCalledTimes(1);
        expect(handlerSaveTodo).toBeCalledWith(
            { 
                "desc": newTodoDesc,
                "done": false, 
                "id": expect.any(Number)
            });
        expect(wrapper.find('[name="todoDesc"]').text()).toEqual("")
        done(); 
    })
 })