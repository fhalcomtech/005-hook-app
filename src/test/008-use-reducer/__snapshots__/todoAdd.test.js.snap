// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TodoAdd Test TodoAdd create snapshot 1`] = `
<div
  className="col-4"
>
  <h2>
    Add Todo
  </h2>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      name="todoDesc"
      onChange={[Function]}
      placeholder="write todo description"
      type="text"
      value=""
    />
    <button
      className="btn btn-success form-control mt-2 ms-0"
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
